core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -134,6 +134,10 @@ public boolean handle(int code) {
             // we've already logged this.
         }
 
+        if (Constants.LINUX) {
```

Comment:
do we need the linux check here, as JNANatives method takes care of that check itself? 
maybe we can have a better name than `maxNumberOfThreads` ? Sorry i dont have a proposed one.

But this could be more consistent with `tryMlockall` and how the other methods here work. They don't check for any OS-specific stuff in this file.